plugins/uefi/fu-uefi-common.c

```diff
@@ -235,6 +238,12 @@ fu_uefi_get_esp_path_for_os (const gchar *base)
 	}
 	if (os_release_id == NULL)
 		os_release_id = "unknown";
+	/* use UEFI removable path as default if the option is set to true */
+	if (fu_device_get_metadata_boolean (device, "RequireRemovablePath")) {
+		esp_path = g_build_filename (base, "EFI", "boot", NULL);
+		if (g_file_test (esp_path, G_FILE_TEST_IS_DIR))
+			return g_steal_pointer (&esp_path);
+	}
```

Comment:
so rather than forcing the path, my suggestion is to put a test at the end after this failed: 
```		
if (g_file_test (id_like_path, G_FILE_TEST_IS_DIR)) {
```

And then that means if an OS path exists it uses that, and if it doesn't it uses removable path if the flag is turned on.